api-catalog-services/src/main/java/com/ca/mfaas/apicatalog/services/cached/CachedApiDocService.java

```diff
@@ -44,16 +44,21 @@ public CachedApiDocService(APIDocRetrievalService apiDocRetrievalService, Transf
      * @return api doc info for the requested service id
      */
     public String getApiDocForService(final String serviceId, final String apiVersion) {
-        String apiDoc = CachedApiDocService.serviceApiDocs.get(new ApiDocCacheKey(serviceId, apiVersion));
-        if (apiDoc == null) {
+        String apiDoc;
+
+        try {
             ApiDocInfo apiDocInfo = apiDocRetrievalService.retrieveApiDoc(serviceId, apiVersion);
-            if (apiDocInfo.getApiDocContent() == null) {
-                return null;
+            if (apiDocInfo == null || apiDocInfo.getApiDocContent() == null) {
+                apiDoc = CachedApiDocService.serviceApiDocs.get(new ApiDocCacheKey(serviceId, apiVersion));
             } else {
                 apiDoc = transformApiDocService.transformApiDoc(serviceId, apiDocInfo);
                 CachedApiDocService.serviceApiDocs.put(new ApiDocCacheKey(serviceId, apiVersion), apiDoc);
             }
+        } catch (Exception e) {
+            log.warn("ApiDoc retrieving problem. {}", e.getMessage());
+            apiDoc = CachedApiDocService.serviceApiDocs.get(new ApiDocCacheKey(serviceId, apiVersion));
```

Comment:
This code `apiDoc = CachedApiDocService.serviceApiDocs.get(new ApiDocCacheKey(serviceId, apiVersion));` duplicated, may be it will be better optimize this? Something like that:
```java
    public String getApiDocForService(final String serviceId, final String apiVersion) {
        String apiDoc = CachedApiDocService.serviceApiDocs.get(new ApiDocCacheKey(serviceId, apiVersion));
        try {
            ApiDocInfo apiDocInfo = apiDocRetrievalService.retrieveApiDoc(serviceId, apiVersion);
            if (apiDocInfo != null && apiDocInfo.getApiDocContent() != null) {
                apiDoc = transformApiDocService.transformApiDoc(serviceId, apiDocInfo);
                CachedApiDocService.serviceApiDocs.put(new ApiDocCacheKey(serviceId, apiVersion), apiDoc);
            }
        } catch (Exception e) {
            log.warn("ApiDoc retrieving problem. {}", e.getMessage());
        }
        return apiDoc;
    }
```

Reply:
Yeap, I think your suggestion is better. Thanks!